backend/test/richard_burton/author_test.exs

```diff
@@ -0,0 +1,77 @@
+defmodule RichardBurton.AuthorTest do
+  @moduledoc """
+  Tests for the Author schema
+  """
+
+  use RichardBurton.DataCase
+
+  alias RichardBurton.Author
+
+  @valid_attrs %{"name" => "J. M. Pereira da Silva"}
+
+  defp changeset(attrs) do
+    Author.changeset(%Author{}, attrs)
+  end
+
+  defp insert(attrs) do
+    attrs |> changeset() |> Repo.insert()
+  end
+
+  defp count do
+    Kernel.length(Author.all())
+  end
+
+  describe "changeset/2" do
+    test "when valid attributes are provided, is valid" do
+      assert changeset(@valid_attrs).valid?
+    end
+
+    test "when name is blank, is invalid" do
+      refute changeset(%{"name" => ""}).valid?
+    end
+
+    test "when name is nil is invalid" do
+      refute changeset(%{"name" => nil}).valid?
+    end
+
+    test "when an author with the provided attributes exist, is invalid" do
+      {:ok, _} = insert(@valid_attrs)
+      {:error, changeset} = insert(@valid_attrs)
+
+      refute changeset.valid?
+      assert %{name: :unique} == Repo.get_errors(changeset)
+    end
+  end
+
+  describe "maybe_insert/1" do
+    test "when there is no author with the provided name, inserts it" do
+      assert 0 == count()
```

Comment:
I think the goal of this test is to check it inserts one record so I wouldn't assume we have zero records when the test starts.

I would do something like:
```elixir
number_of_authors = count()

author = Author.maybe_insert!(@valid_attrs)

assert [^author] = Author.all()
assert number_of_authors + 1 == count()
```

However, `[^author] = Author.all()` is already checking there's only one so I'm not sure it's even relevant to check again.

Reply:
Makes sense, I made some adjustments to the tests and the `count` function may not be necessary anymore. LMK what you think.